### PR TITLE
Adding help pages

### PR DESCRIPTION
**Pull request by humdingerb**
Tell me if it's OK like this or where I got the syntax etc. wrong...

* I put the screenshots under _images/help/_.
* The "up.png" is currently in _images/_, maybe it'll be used on other pages.
* index.twig: I'm unsure if I had to add the links to the developer and user help pages. And if so, if that is the right way to link pages, `{{ link_to_route('help.user', 'For the translators') }}` instead of `<a href>`).

I see that my brilliant Haiku didn't make it in... oh well. :)

What about CSS? Is inline "style:..." allowed?
May I fiddle with the public/css/style.css? The <h> are a bit large...

### CODE REVIEW

File: resources/views/help/developer.twig [hunk 1]
```diff
@@ -0,0 +1,127 @@
+{% extends "layouts.master" %}
+
+{% block title %}Help for developers{% endblock %}
+
+{% block content %}
+<ol class="breadcrumb">
+    <li>{{ link_to_route('projects.index', 'Home') }}</li>
+    <li class="active">Help for developers</li>
```
Comment (KapiX) [inline resources/views/help/developer.twig]: I'd split this up to "Help / Developers" where Help is a link to main Help page (route 'help').

File: resources/views/help/index.twig [hunk 2]
```diff
@@ -9,8 +9,18 @@
...
 <p>Polyglot was conceived and implemented by Kacper Kasper.</p>
 <p>It aims to be a very quick and easy tool for Haiku application developers to publish new and updated catkeys of their projects. Polyglot allows users to browse these projects and quickly localize the strings still missing from their native language.</p>
 <p>If you find bugs, have feature requests, or would like to request adding another language for translation, please file a <a href="https://github.com/KapiX/Polyglot/issues">ticket at the issue tracker</a>.</p>
-<p>Thanks to Humdinger for providing ideas and artwork.</p>
+<p>Thanks to Humdinger for providing ideas, documentation and artwork.</p>
+
+<h2>Help</h2>
+
+<ol type="A">
+	<li>{{ link_to_route('help.user', 'For the translators') }} - Read up on using Polyglot and some tips on localization in general.</li>
```
Comment (KapiX) [inline resources/views/help/index.twig]: `link_to_route('help', 'For the translators', {article: 'user'})`

File: resources/views/help/developer.twig [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	<li><a href="#adding">Adding a project</a></li>
+	<li><a href="#uploading">Uploading existing xx.catkeys (translations)</a></li>
+	<li><a href="#updating">Updating en.catkeys</a></li>
+	<li><a href="#exporting">Exporting translated xx.catkeys</a></li>
+	<li><a href="#tips">Tips</a></li>
+	<li><a href="#tldr">TL;DR</a></li>
+</ul>
+
+<h2>
+<a href="#"><img src="images/up.png" style="border:none;float:right" alt="index" /></a>
```
Comment (KapiX) [inline resources/views/help/developer.twig]: If you need a link use `{{ url('images/up.png') }}`. I'd also prefer if styles were in style.css.

File: resources/views/help/developer.twig [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+{% block title %}Help for developers{% endblock %}
+
+{% block content %}
+<ol class="breadcrumb">
+    <li>{{ link_to_route('projects.index', 'Home') }}</li>
+    <li class="active">Help for developers</li>
+</ol>
+
+<h2>Help and guidance for developers</h2>
```
Comment (KapiX) [inline resources/views/help/developer.twig]: This can be removed I think, it's right under breadcrumbs which say what page it is.

File: resources/views/help/developer.twig (hunk 3 above)
Comment (humdingerb) [reply resources/views/help/developer.twig]: I'm not 100% positive I do that right. I added this to style.css:
```
img.centered {
	margin-bottom: 10px;
}

img.float {
	border: none; 
	float: right;
}
```
How would I assign the 'float' style to the "up" image (or the 'centered' to html_image('images/help/user-startpage.png')?

```<a href="#">index</a>``` ?

Comment (KapiX) [reply resources/views/help/developer.twig]: It would be even better if you used {{ html_image }} for up images as well.

Also change class names to be more descriptive, like *up* and *screenshot*.

See below:
```twig
{{ html_image('images/help/dev-upload-xx-catkeys.png', 'Uploading the en.catkeys', {'class': 'center-block'}) }}
```

Comment (humdingerb) [reply resources/views/help/developer.twig]: OK. Like that?
```<a href="#">{{ html_image('images/up.png', 'Go up', {'class': 'up'}) }}</a>```

Comment (KapiX) [reply resources/views/help/developer.twig]: Exactly.

File: resources/views/help/developer.twig [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+{{ html_image('images/help/dev-upload-xx-catkeys.png', 'The "Translation" page', {'class': 'screenshot'}) }}
+
+<p>At the top, you can now upload the <i>xx.catkeys</i> file for that language.<br />
+When you read the info box, you see that you now have to un-set the <i>Needs work</i> checkbox and click <i>Save</i> for all strings that didn't need translation, because they are the same as the original English. Depending on the language, this is quite common e.g. for <i>OK</i> buttons or the name of the application.</p>
+
+<p>Uploading translated <i>xx.catkeys</i> has no effect on your <i>en.catkeys</i>. Changes always go from the <i>en.catkeys</i> to the translations, never the other way around.<br />
+If the uploaded <i>xx.catkeys</i> are outdated and contain additional, now obsolete strings, those strings are simply discarded. Only strings that have the counterpart of their English original in the <i>en.catkeys</i> in Polyglot's database are imported.</p>
+
+<h2 class="help">
+<a href="#">{{ url('<a href="#">{{ html_image('images/up.png', 'Go up', {'class': 'go-up'}) }}</a>
```
Comment (KapiX) [inline resources/views/help/developer.twig]: Whoa, what happened here? ;)

File: resources/views/help/developer.twig [hunk 6]
```diff
@@ -0,0 +1,127 @@
+{% extends "layouts.master" %}
+
+{% block title %}Help for developers{% endblock %}
+
+{% block content %}
+<ol class="breadcrumb">
+    <li>{{ link_to_route('projects.index', 'Home') }}</li>
+    <li>{{ link_to_route('help', 'Help') }}</li>
+    <li class="active">Help for developers</li>
```
Comment (KapiX) [inline resources/views/help/developer.twig]: Just `Developers`

File: resources/views/help/index.twig [hunk 7]
```diff
@@ -7,10 +7,20 @@
     <li>{{ link_to_route('projects.index', 'Home') }}</li>
     <li class="active">Help</li>
 </ol>
-{{ html_image('images/Polyglot_100px.png', 'Polyglot logo', {'class': 'center-block'}) }}
+{{ html_image('images/Polyglot_100px.png', 'Polyglot logo', {'class': 'screenshot'}) }}
```
Comment (KapiX) [inline resources/views/help/index.twig]: Revert this change, the logo doesn't need bottom margin.

File: resources/views/help/user.twig [hunk 8]
```diff
@@ -0,0 +1,76 @@
+{% extends "layouts.master" %}
+
+{% block title %}Help for translators{% endblock %}
+
+{% block content %}
+<ol class="breadcrumb">
+    <li>{{ link_to_route('projects.index', 'Home') }}</li>
+    <li>{{ link_to_route('help', 'Help') }}</li>
+    <li class="active">Help for translators</li>
```
Comment (KapiX) [inline resources/views/help/user.twig]: Just `Translators`

File: public/css/style.css [hunk 9]
```diff
@@ -47,3 +47,31 @@ a.navbar-brand {
 select#languages {
     height: 30em;
 }
+
+h2 {
```
Comment (KapiX) [inline public/css/style.css]: These (h2 and h3) are messing with headers everywhere. Please remove these.

For reference, this is what is used on the site:
https://getbootstrap.com/docs/3.3/css/
https://getbootstrap.com/docs/3.3/components/

File: public/css/style.css [hunk 10]
```diff
@@ -47,3 +47,31 @@ a.navbar-brand {
...
+	font-weight: normal;
+	color: #000000;
+}
+h3 {
+	font-size: 1.2em;
+	font-weight: normal;
+	color: #000000;
+}
+
+h2.help {
```
Comment (KapiX) [inline public/css/style.css]: This looks out of place. Stick with bootstrap styles.

File: public/css/style.css [hunk 11]
```diff
@@ -47,3 +47,31 @@ a.navbar-brand {
...
+	font-size: 1.3em;
+	font-weight: normal;
+	color: #0c3762;
+	border-bottom: dotted thin #e0e0e0;
+	margin-top: 25px;
+}
+
+img.screenshot {
```
Comment (KapiX) [inline public/css/style.css]: Actually, remove this style. Wrap images in <p> so they get the same styling.

File: public/css/style.css (hunk 9 above)
Comment (humdingerb) [reply public/css/style.css]: OK. I'd have hoped that this would show the "Add project" etc. stuff not so very huge as it is now...

Comment (humdingerb) [reply public/css/style.css]: OK. I'd have hoped that this would show the "Add project" etc. stuff not so very huge as it is now...

File: public/css/style.css (hunk 10 above)
Comment (humdingerb) [reply public/css/style.css]: OK. I do like those lines under the headings, though, as they don't have the go-up buttons look like they are hanging in the air. :)

File: public/css/style.css (hunk 11 above)
Comment (humdingerb) [reply public/css/style.css]: OK. p.screenshot it is, adding <p.screenshot> around sreenshot images.

File: public/css/style.css [hunk 12]
```diff
@@ -40,10 +40,19 @@ p.comment {
     font-style: italic;
 }
 
+p.screenshot {
```
Comment (KapiX) [inline public/css/style.css]: No, p has margin-bottom:10px by default, that is not necessary.

File: resources/views/help/developer.twig [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+<a id="adding" name="adding">Adding a project</a></h2>
+
+<p>Simply enter the new project name into the text box of the Polyglot start page, click <i>Add project</i> and it'll appear in the list of projects. Click on it and you get to the "<i>Files</i>" page of your project:</p>
+
+{{ html_image('images/help/dev-files-name.png', 'The "Files" page to name parts of your application that need their catkeys translated') }}
+
+<p>After adding the identifying name for this main part of your app (by default named "Application"), you upload the <i>en.catkeys</i> for it.<br />
+You'll see the signature of the application and the fingerprint, that changes every time you create a new <i>en.catkeys</i> with changed GUI strings:</p>
+
+<p class="screenshot>{{ html_image('images/help/dev-upload-xx-catkeys.png', 'Uploading the en.catkeys') }}
```
Comment (KapiX) [inline resources/views/help/developer.twig]: Missing closing tag.

File: resources/views/help/developer.twig [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+
+<p>When all the languages you need are 100% finished, you can export and download their <i>xx.catkeys</i> to use when building your application.</p>
+
+<h2 class="help">
+<a href="#">{{ html_image('images/up.png', 'Go up', {'class': 'go-up'}) }}</a>
+<a id="adding" name="adding">Adding a project</a></h2>
+
+<p>Simply enter the new project name into the text box of the Polyglot start page, click <i>Add project</i> and it'll appear in the list of projects. Click on it and you get to the "<i>Files</i>" page of your project:</p>
+
+{{ html_image('images/help/dev-files-name.png', 'The "Files" page to name parts of your application that need their catkeys translated') }}
```
Comment (KapiX) [inline resources/views/help/developer.twig]: Missing p

File: resources/views/help/developer.twig [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+
+<p>At the top, you can now upload the <i>xx.catkeys</i> file for that language.<br />
+When you read the info box, you see that you now have to un-set the <i>Needs work</i> checkbox and click <i>Save</i> for all strings that didn't need translation, because they are the same as the original English. Depending on the language, this is quite common e.g. for <i>OK</i> buttons or the name of the application.</p>
+
+<p>Uploading translated <i>xx.catkeys</i> has no effect on your <i>en.catkeys</i>. Changes always go from the <i>en.catkeys</i> to the translations, never the other way around.<br />
+If the uploaded <i>xx.catkeys</i> are outdated and contain additional, now obsolete strings, those strings are simply discarded. Only strings that have the counterpart of their English original in the <i>en.catkeys</i> in Polyglot's database are imported.</p>
+
+<h2 class="help">
+<a href="#">{{ html_image('images/up.png', 'Go up', {'class': 'go-up'}) }}</a>
+<a id="updating" name="updating">Updating en.catkeys</a></h2>
```
Comment (KapiX) [inline resources/views/help/developer.twig]:
```twig
<a id="updating" name="updating">Updating en.catkeys</a>
```
better:
```twig
<a id="updating" name="updating"></a>Updating en.catkeys
```

Right now headers have different color and underline on hover.

**Comment by KapiX** [comment]
Thanks!
